test/scalability/mzcompose.py

```diff
@@ -0,0 +1,262 @@
+
+    init_conn = endpoint.sql_connection()
+    init_conn.autocommit = True
+    init_cursor = init_conn.cursor()
+    for init_sql in init_sqls:
+        print(init_sql)
+        init_cursor.execute(init_sql)
+
+    connect_sqls = schema.connect_sqls()
+
+    print(
+        f"Creating a cursor pool with {concurrency} entries against endpoint: {endpoint.url()}"
+    )
+    cursor_pool = []
+    for i in range(concurrency):
+        conn = endpoint.sql_connection()
+        conn.autocommit = True
+        cursor = conn.cursor()
+        for connect_sql in connect_sqls:
+            cursor.execute(connect_sql)
+        cursor_pool.append(cursor)
+
+    print(f"Benchmarking workload {type(workload)} at concurrency {concurrency} ...")
+    operations = workload.operations()
+
+    start = time.time()
+    with futures.ThreadPoolExecutor(concurrency) as executor:
+        measurements = executor.map(
+            execute_operation,
+            [
+                (
+                    workload,
+                    concurrency,
+                    operations[i % len(operations)],
+                    cursor_pool[i % concurrency],
+                    i,
+                    i % concurrency,
+                )
+                for i in range(count)
+            ],
+        )
+    wallclock_total = time.time() - start
+
+    df_detail = pd.DataFrame()
+    for measurement in measurements:
+        df_detail = pd.concat([df_detail, pd.DataFrame([measurement])])
+    print("Best and worst individual measurements:")
+    print(df_detail)
+
+    print(
+        f"concurrency: {concurrency}; wallclock_total: {wallclock_total}; tps = {count/wallclock_total}"
+    )
+
+    df_total = pd.DataFrame(
+        [
+            {
+                "concurrency": concurrency,
+                "wallclock": wallclock_total,
+                "workload": type(workload).__name__,
+                "count": count,
+                "tps": count / wallclock_total,
+            }
+        ]
+    )
+
+    return (df_total, df_detail)
+
+
+def run_workload(
+    c: Composition,
+    args: argparse.Namespace,
+    endpoint: Endpoint,
+    schema: Schema,
+    workload: Workload,
+) -> None:
+    df_totals = pd.DataFrame()
+    df_details = pd.DataFrame()
+    for concurrency in range(
+        args.min_concurrency, args.max_concurrency + 1, args.concurrency_step
+    ):
+        df_total, df_detail = run_with_concurrency(
+            c, endpoint, schema, workload, concurrency, args.count
+        )
+        df_totals = pd.concat([df_totals, df_total])
+        df_details = pd.concat([df_details, df_detail])
+
+        df_totals.to_csv(f"results/{type(workload).__name__}.csv")
+        df_details.to_csv(f"results/{type(workload).__name__}_details.csv")
+
+
+def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
+    parser.add_argument(
+        "--target", help="Target for the benchmark: 'HEAD', 'local', or a DockerHub tag"
```

Comment:
The doc mentions further options (e.g., `postgres`)

Reply:
Fixed.